config/v1/types_authentication.go

```diff
@@ -253,9 +254,16 @@ type TokenIssuer struct {
 	// The Kubernetes API server determines how authentication tokens should be handled
 	// by matching the 'iss' claim in the JWT to the issuerURL of configured identity providers.
 	//
-	// issuerURL must use the 'https' scheme.
+	// Must be at least 1 character and must not exceed 267 characters in length.
+	// Must be a valid URL that uses the 'https' scheme and does not contain a query, fragment or user.
 	//
-	// +kubebuilder:validation:Pattern=`^https:\/\/[^\s]`
+	// +kubebuilder:validation:XValidation:rule="(oldSelf.hasValue() && oldSelf.orValue(\"\") == self) || isURL(self)",message="must be a valid URL",optionalOldSelf=true
```

Comment:
I was under the impression that CEL rules would ratchet by themselves without needing specific ratcheting to be written, have you tested these? If you have, can we test without the custom ratcheting in place?

Reply:
I did test these without the custom ratcheting in place and consistently got failures

Reply:
Maybe this is because I need to do an `isURL` before a `url`?  🤔